general/mem_manager.cpp

```diff
@@ -1130,14 +1130,13 @@ void MemoryManager::InsertAlias(const void *base_ptr, void *alias_ptr,
                             internal::Alias{&mem, offset, bytes, 1, mem.h_mt});
    if (res.second == false) // alias_ptr was already in the map
    {
-      if (res.first->second.mem != &mem || res.first->second.offset != offset)
-      {
-         mfem_error("alias already exists with different base/offset!");
-      }
-      else
-      {
-         res.first->second.counter++;
-      }
+      internal::Alias &alias = res.first->second;
+      // Update the alias data in case the existing alias is dangling
+      alias.mem = &mem;
+      alias.offset = offset;
+      alias.bytes = bytes; // aliases with different sizes should be fine
+      alias.h_mt = mem.h_mt;
+      alias.counter++;
```

Comment:
We are updating one `alias` found in the map that is `dangling` by reassigning value, but using the counter of the old alias. What happens to the code that is still using this pointer? Do we want it to 'continue' to progress with some new data ?

Reply:
The dangling alias is still registered and we want to let it naturally un-register itself when deleted. Of course, using the actual memory of such dangling alias will be wrong but that is the same bug that you will see if using simply pointers instead of `Memory` objects. If we want to detect and report such misuse, we'll need bigger changes, I think.